lib/octobox/configurator.rb

```diff
@@ -1,23 +1,30 @@
 module Octobox
-  class << self
+  class Configurator
+    def github_domain
+      @github_domain || ENV.fetch('GITHUB_DOMAIN', 'https://github.com')
+    end
+    attr_writer :github_domain
+
+    def github_api_prefix
+      return @github_domain_api_prefix if defined?(@github_domain_api_prefix)
+
+      if github_domain != 'https://github.com'
+        @github_domain_api_prefix = "#{github_domain}/api/v3"
+      else
+        @github_domain_api_prefix = "https://api.github.com"
+      end
+    end
```

Comment:
Review these two methods carefully; they were not configurable beyond env before (due to initializer timing concerns) so they're slightly edited. If I understand correctly the api_prefix is entirely dependent on the domain so there's no writer here intentionally.